nanoFramework.Tools.DebugLibrary.Shared/PortSerial/DeviceWatcher.cs

```diff
@@ -99,63 +99,23 @@ public List<string> GetPortNames()
 
         private List<string> GetPortNames_Linux()
         {
-            const string sysTtyDir = "/sys/class/tty";
-            const string sysUsbDir = "/sys/bus/usb-serial/devices/";
-            const string devDir = "/dev/";
+            List<string> ports = new List<string>();
 
-            if (Directory.Exists(sysTtyDir))
+            string[] ttys = System.IO.Directory.GetFiles("/dev/", "tty*");
+            foreach (string dev in ttys)
             {
```

Comment:
I believe you could simplify it with a LINQ expression as it follows:

`
            var ports = matches.Where(dev =>
                                ttys.All(m => m.StartsWith(dev, StringComparison.InvariantCultureIgnoreCase)))
                            .ToList();`

Reply:
not super sure Linq does simplify anything :-D Feels like it brings more complexity on how you can read the code.